fe/fe-core/src/main/java/com/starrocks/sql/common/SqlDigestBuilder.java

```diff
@@ -50,6 +92,26 @@ public String visitInPredicate(InPredicate node, Void context) {
             }
         }
 
+        @Override
+        public String visitCompoundPredicate(CompoundPredicate node, Void context) {
+            Pair<Integer, Integer> pair = countCompound(node);
+            if (pair.first + pair.second >= MASSIVE_COMPOUND_LIMIT) {
+                // Only record de-duplicated slots if there are too many compounds
+                Set<SlotRef> exprs = Sets.newHashSet();
+                traverse(node, x -> {
```

Comment:
why not use `node.collectAllSlotRefs()` directly?

Reply:
well, I should use it. but it seems I should improve the performance of that method first.

Reply:
it constructs a lot of intermediate list,, i'm afraid the performance can be terrible